starfish/imagestack/imagestack.py

```diff
@@ -175,35 +159,39 @@ def __init__(
             dims=data_dimensions,
             coords=data_tick_marks,
         )
-        self._coordinates = xr.DataArray(
-            np.empty(
-                shape=coordinates_shape,
-                dtype=np.float32,
-            ),
-            dims=coordinates_dimensions,
-            coords=coordinates_tick_marks,
-        )
 
         all_selectors = list(self._iter_axes({Axes.ROUND, Axes.CH, Axes.ZPLANE}))
         first_selector = all_selectors[0]
         tile = tile_data.get_tile(r=first_selector[Axes.ROUND],
                                   ch=first_selector[Axes.CH],
                                   z=first_selector[Axes.ZPLANE])
-        # only compare X,Y coords
+
+        # Set up coordinates
+        self._data[Coordinates.X.value] = xr.DataArray(
+            np.linspace(tile.coordinates[Coordinates.X][0], tile.coordinates[Coordinates.X][1],
+                        self.xarray.sizes[Axes.X.value]), dims=Axes.X.value)
+        self._data[Coordinates.Y.value] = xr.DataArray(
+            np.linspace(tile.coordinates[Coordinates.Y][0], tile.coordinates[Coordinates.Y][1],
+                        self.xarray.sizes[Axes.Y.value]), dims=Axes.Y.value)
+        if Coordinates.Z in tile.coordinates:
+            # Fill with zeros for now, then replace with calculated midpoints
```

Comment:
Is there no way to look up this information? would it make sense to record the extrema on the TileSet?

Reply:
which info are you talking about?

Reply:
@ambrosejcarr

Reply:
The Z mid-point information.